vincepub/serializers.py

```diff
@@ -136,3 +140,134 @@ class Meta:
+    product_tree = serializers.SerializerMethodField('get_csafprods')
+    mproduct_tree = {"branches": []}
+    template_json_dir = os.path.join(os.path.dirname(__file__),
+                                     '..','vinny','templatesjson', 'csaf')
+
+    class Meta:
+        model = VUReport
+        fields = ["document","vulnerabilities","product_tree"]
+
+    def get_csafdocument(self,vr):
+        tfile = os.path.join(self.template_json_dir,"document.json")
+        if not os.path.exists(tfile):
+            return {"error": "Template file for csaf missing"}
+        csafdocument_template = open(tfile,"r").read()
+        vulnote = reverse("vincepub:vudetail", args=[vr.idnumber])
+        publicurl = f"{settings.KB_SERVER_NAME}{vulnote}"
+        ackurl = f"{settings.KB_SERVER_NAME}{vulnote}#acknowledgements"
+        case_status = "final"
+
+        if vr.dateupdated:
+            revision_date = vr.dateupdated
+            revision_number = vr.dateupdated.strftime("1.%Y%m%d%H%M%S")
+        else:
+            revision_date = datetime.datetime.now()
+            revision_number = revision_date.strftime("1.%Y%m%d%H%M%S")
+        if vr.revision:
+            revision_number = revision_number + "." + str(vr.revision)
+        else:
+            revision_number = revision_number + ".0"
+        case_version = revision_number
+        csafdocument = csafdocument_template % {
+            "publicurl": publicurl,
+            "ackurl": ackurl,
+            "summary": json.dumps(vr.overview),
+            "LEGAL_DISCLAIMER": settings.LEGAL_DISCLAIMER,
+            "title": json.dumps(vr.name),
+            "due_date": vr.publicdate,
+            "VINCE_VERSION": settings.VERSION,
+            "vu_vuid": vr.vuid,
+            "revision_date": revision_date,
+            "revision_number": revision_number,
+            "case_status": case_status,
+            "case_version": case_version
+        }
+        csafdoc = json.loads(csafdocument, strict=False)
+        if vr.vulnote.references and vr.vulnote.references != '':
+            refs = re.split('\r?\n',vr.vulnote.references)
+            csafdoc["references"] += list(map(lambda x: {"url": x, "summary": x},refs))
+        vens = list(Vendor.objects.filter(note=vr.vulnote))
+        for ven in vens:
+            if ven.statement:
+                veninfo = {"category": "other",
+                           "text": ven.statement,
+                           "title": f"Vendor statment from {ven.vendor}"}
+                csafdoc["notes"] += [veninfo]
+            if ven.references:
+                refs = re.split('\r?\n',ven.references)
+                csafdoc["references"] += list(map(lambda x: {"url": x, "summary": f"Reference(s) from vendor \"{ven.vendor}\""},refs))
+            if ven.addendum:
+                addinfo = {"category": "other",
+                           "text": ven.addendum,
+                           "title": "CERT/CC comment on {ven.vendor} notes"}
```

Comment:
This should be parameterized in the open source release.

Reply:
Thanks for your input.. More to fix on this issue, so overriding some of your requested changes and testing our GitHub repo.

Vijay